Feat(eos_validate_state): Added the validation for IP security connections (#3911)
aristanetworks/ansible-avd: ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md

- (New) AvdTestAPIHttpsSSL (No Ansible tags, use the new `skipped_tests` variable instead)
  - VerifyAPIHttpsSSL: Validate eAPI HTTPS SSL profile status.

> - (New) VerifySpecificIPSecConn (No Ansible tags, use the new `skipped_tests` variable instead)

Reviewer: ```suggestion
- (New) AvdTestIPSecurity (No Ansible tags, use the new `skipped_tests` variable instead)
```

>   - VerifySpecificIPSecConn: Validate the IP security connection status.

Reviewer: Can we add more details here about the test here? Users reading this doc should be able to easily figure out what the test is about.

>   - VerifySpecificIPSecConn: Validate the IP security connections for a peer in VRF are established.

Reviewer 2: ```suggestion
  - VerifySpecificIPSecConn: Validate the IP security connections for a peer in a given VRF are established.
```

Reviewer 2: Hmmm docstring says only default is supported so maybe we should mention this

>   - VerifySpecificIPSecConn: Validates the establishment of IP security connections for a peer within the default VRF. This test exclusively examines IPsec connections defined in the router's path selection configuration.

Reviewer 2: ```suggestion
  - VerifySpecificIPSecConn: Validates the establishment of IP security connections for a peer within the default VRF. In its current state, the test validates only IPsec connections defined as static peers under the `router path-selection` section of the configuration.
```
